packages/foam-vscode/static/graphs/default/graph.js

```diff
@@ -97,6 +97,12 @@ const Actions = {
       });
       m.data.links = links; // links can be swapped out without problem
 
+      // check that selected/hovered nodes are still valid (see #397)
+      m.hoverNode = remaining.has(m.hoverNode) ? m.hoverNode : null;
+      m.selectedNodes = new Set(
+        Array.from(m.selectedNodes).filter(nId => remaining.has(nId))
+      );
+
```

Comment:
I rebased my branch for #394 on top of this -- I'm noticing that when I save a file the graph loses track of which node is selected / being hovered over. Could that be related to my change of not deleting the node anymore on save?

Reply:
I just reproduced this on master, so it's unrelated to your changes - I am investigating what's causing this

Reply:
if you rebase from master the issue is now fixed

Reply:
Thanks!